src/test/fixtures/utils.js

```diff
@@ -282,3 +282,41 @@ export function findFillTextPositionFromDrawLog(
 
   return positions[0];
 }
+
+/**
+ * JSDom only sends one event at a time, but a lot of component logic assumes that
+ * events come in a natural cascade. This utility ensures that cascasde gets fired
+ * correctly. This also includes the fix to make properties like pageX work.
+ */
+export function fireFullClick(
+  element: HTMLElement,
+  options?: FakeMouseEventInit
+) {
+  fireEvent(element, getMouseEvent('mousedown', options));
+  fireEvent(element, getMouseEvent('mouseup', options));
+  fireEvent(element, getMouseEvent('click', options));
+}
+
+export function fireFullContextMenu(
+  element: HTMLElement,
+  options?: FakeMouseEventInit
+) {
+  const isMacContextMenu =
+    options &&
+    options.ctrlKey &&
+    !options.metaKey &&
+    !options.shiftKey &&
+    !options.altKey;
+
+  if (!isMacContextMenu) {
+    options = {
+      ...(options || {}),
+      button: 2,
+      buttons: 2,
+    };
+  }
```

Comment:
It's not especially clear to me what you're trying to do here.
I would expect rather a boolean parameter that would change the options to add `ctrlKey` and change the buttons, rather than infering this from the presence of `ctrlKey`. Then tests would be clearer:
```js
fireFullContextMenu(element, { foo }, { useMacOsCtrlKey: true });
```
(I'm not sure for the property name, you may have a better idea).

What do you think?

Reply:
Hmm, I think I'd prefer to avoid the config, and just allow through valid configurations. I updated my code a bit and added clarifying comments as to the strategy. I hope this addresses the: "It's not especially clear to me what you're trying to do here."